_includes/pbc/all/install-features/202304.0/marketplace/install-the-marketplace-product-offer-feature.md

```diff
@@ -0,0 +1,1822 @@
+     */
+    public function getEventSubscriberCollection(): EventSubscriberCollectionInterface
+    {
+        $eventSubscriberCollection = parent::getEventSubscriberCollection();
+        $eventSubscriberCollection->add(new MerchantSearchEventSubscriber());
+        $eventSubscriberCollection->add(new MerchantProductOfferSearchEventSubscriber());
+
+        return $eventSubscriberCollection;
+    }
+}
+```
+
+Register the synchronization queue and synchronization error queue:
+
+**src/Pyz/Client/RabbitMq/RabbitMqConfig.php**
+
+```php
+<?php
+
+namespace Pyz\Client\RabbitMq;
+
+use Spryker\Client\RabbitMq\RabbitMqConfig as SprykerRabbitMqConfig;
+use Spryker\Shared\ProductOfferStorage\ProductOfferStorageConfig;
+
+class RabbitMqConfig extends SprykerRabbitMqConfig
+{
+    /**
+     * @return list<string>
+     */
+    protected function getSynchronizationQueueConfiguration(): array
+    {
+        return [
+            ProductOfferStorageConfig::PRODUCT_OFFER_SYNC_STORAGE_QUEUE,
+        ];
+    }
+
+}
+```
+
+#### Configure message processors
+
+| PLUGIN                                            | SPECIFICATION                                                                                                    | PREREQUISITES | NAMESPACE                                              |
+|---------------------------------------------------|------------------------------------------------------------------------------------------------------------------|---------------|--------------------------------------------------------|
+| SynchronizationStorageQueueMessageProcessorPlugin | Configures all merchant product offers to sync with Redis storage and marks messages as failed in case of error. |               | Spryker\Zed\Synchronization\Communication\Plugin\Queue |
+
+**src/Pyz/Zed/Queue/QueueDependencyProvider.php**
+
+```php
+<?php
+
+namespace Pyz\Zed\Queue;
+
+use Spryker\Shared\ProductOfferStorage\ProductOfferStorageConfig;
+use Spryker\Zed\Kernel\Container;
+use Spryker\Zed\Queue\QueueDependencyProvider as SprykerDependencyProvider;
+use Spryker\Zed\Synchronization\Communication\Plugin\Queue\SynchronizationStorageQueueMessageProcessorPlugin;
+
+class QueueDependencyProvider extends SprykerDependencyProvider
+{
+    /**
+     * @param \Spryker\Zed\Kernel\Container $container
+     *
+     * @return array<string, \Spryker\Zed\Queue\Dependency\Plugin\QueueMessageProcessorPluginInterface>
+     */
+    protected function getProcessorMessagePlugins(Container $container): array
+    {
+        return [
+            ProductOfferStorageConfig::PRODUCT_OFFER_SYNC_STORAGE_QUEUE => new SynchronizationStorageQueueMessageProcessorPlugin(),
+        ];
+    }
+}
+```
+
+#### Set up, re-generate, and re-sync features
+
+| PLUGIN                                                              | SPECIFICATION                                                       | PREREQUISITES | NAMESPACE                                                            |
```

Comment:
We have more than one step in this section. Add the intro step here. For example:
```suggestion
1. Set up the following plugins:

| PLUGIN                                                              | SPECIFICATION                                                       | PREREQUISITES | NAMESPACE                                                            |
```